app/src/main/java/org/mozilla/focus/session/SessionManager.java

```diff
@@ -260,4 +260,17 @@ public void removeCurrentSession() {
 
         this.sessions.setValue(sessions);
     }
+
+    // Replaces a (custom-tab) session by its no custom-tab version
+    public void replaceSessionWithNoCustomTab(Session session) {
+        if (!session.isCustomTab()) {
+            return;
+        }
+
+        Session noCustomTabSession = new Session(Source.MENU, session.getUrl().getValue());
+        noCustomTabSession.setBlockingEnabled(session.isBlockingEnabled());
+
+        this.addSession(noCustomTabSession);
+        this.removeSession(session.getUUID());
```

Comment:
Here I am forced to add the new session before removing the custom-tab session otherwise we have this exception:

![image](https://user-images.githubusercontent.com/19215803/34659136-4094c65e-f42d-11e7-8b9b-678bb5149627.png)

Reply:
I'm not sure why this would be throwing `IllegalStateException` – I'd expect `NullPointerException`. I also can't find where such an IllegalStateException would be thrown from.

I'm guessing that maybe we don't handle the case where all sessions are removed in a custom tab session properly.

It's best if we can figure out why this is happening but if you've already looked for a little bit and can't find anything, I think reversing the logic order seems sound and I'm okay landing it without knowing exactly why this is happening.